### PR TITLE
Serve, run, and compile a subset of items

### PR DESCRIPTION
**Pull request by tgsoverly**
This filters the traces that are run and compiled for any node with a name.  Serve was more work to support because it is a local serve that works on the project json, I only implemented the dashboard level filtering for now.

Simple Project: `visivo serve -nf "Simple Dashboard"`

### CODE REVIEW

File: tests/models/test_project.py [hunk 1]
```diff
@@ -24,25 +23,6 @@ def test_Project_simple_data():
     assert project.name == "development"
 
 
-def test_Project_filter_trace():
```
Comment (tgsoverly) [inline tests/models/test_project.py:27]: Removed this old filter

File: visivo/commands/serve_phase.py [hunk 2]
```diff
@@ -12,14 +12,22 @@
...
     project_json = ""
     with open(f"{output_dir}/project.json", "r") as f:
         project_json = json.load(f)
+    
+    if name_filter:
+        dashboards = [d for d in project_json['dashboards'] if d["name"] == name_filter]
+        if len(dashboards) == 1:
+            project_json["dashboards"] = dashboards
+        else:
+            raise click.ClickException(f"Currently the serve command name filtering only supports filtering at the dashbaord level.  No dashboard with {name_filter} found.")
```
Comment (jpjsonic) [inline visivo/commands/serve_phase.py:25]: 👍 Nice this a good error.

File: visivo/commands/serve_phase.py [hunk 3]
```diff
@@ -12,14 +12,22 @@
...
 
 
-def get_project_json(output_dir):
+def get_project_json(output_dir, name_filter=None):
     project_json = ""
     with open(f"{output_dir}/project.json", "r") as f:
         project_json = json.load(f)
+    
+    if name_filter:
+        dashboards = [d for d in project_json['dashboards'] if d["name"] == name_filter]
```
Comment (jpjsonic) [inline visivo/commands/serve_phase.py:21]: I wonder if there's way for us to read the project from the json back into a project object that has a dag so we can use the same filtering that you generated for the compile & run phase.

**Review by jpjsonic — APPROVED** [review]
Looks great!

File: visivo/commands/serve_phase.py (hunk 3 above)
Comment (tgsoverly) [reply visivo/commands/serve_phase.py:21]: We could do that for sure, might be a bit slower.

Comment (jpjsonic) [reply visivo/commands/serve_phase.py:21]: Ah yeah, maybe after we address import performance that will be more practical

Comment (tgsoverly) [reply visivo/commands/serve_phase.py:21]: Also, I just had some business questions about what would happen if you served a trace, for example

**Pull request merged**